README.md

```diff
@@ -183,16 +183,69 @@ torch.testing.assert_close(
 )
 ```
 
+### Lazy preallocation
+
+Pre-allocating tensors can be cumbersome and hard to scale if the list of preallocated
+items varies according to the script configuration. TensorDict solves this in an elegant way.
+Assume you are working with a function `foo() -> TensorDict`, e.g.
+```python
+def foo():
+    tensordict = TensorDict({}, [])
+    tensordict["a"] = torch.randn(3)
+    tensordict["b"] = TensorDict({"c": torch.zeros(2)}, [])
+    return tensordict
+```
+and you would like to call this function repeatedly. You could do this in two ways.
+The first would simply be to stack the calls to the function:
+```python
+tensordict = torch.stack([foo() for _ in range(N)])
+```
+However, you could also choose to preallocate the tensordict:
+```python
+tensordict = TensorDict({}, [N])
+for i in range(N):
+    tensordict[i] = foo()
```

Comment:
Nitpick: The syntax is a bit confusing. `tensordict = TensorDict({}, [N])` looks like a list of tensordicts :) Not sure what we can change but sharing the perspective as a new user.